### PR TITLE
Set-GitHubRepository: Add Support for 'Secret Scanning' Option

### LINKED ISSUE #390: Set-GitHubRepository: Add Support for 'Secret Scanning' Option
#### Feature Idea Summary

GitHub have recently made Secret Scanning available for free on public repositories. The ability to enable this should be added to the `Set-GitHubRepository` Cmdlet.

Note: this can't be added to the `New-GitHubRepository` cmdlet as the underlying GitHub create repo API does not support it.

##### References:

- [Secret scanning is now available for free on public repositories](https://github.blog/changelog/2022-12-15-secret-scanning-is-now-available-for-free-on-public-repositories/)
- [About secret scanning](https://docs.github.com/en/code-security/secret-scanning/about-secret-scanning)

##### API Reference

- https://docs.github.com/en/rest/repos/repos#update-a-repository

##### Parameters

- `security_and_analysis.secret_scanning.status`

#### Requested Assignment

- If possible, I would like to implement this.

### PR DESCRIPTION
**Pull request by X-Guardian**
#### Description

This PR adds the `SecretScanning` parameter to the `Set-GitHubRepository` function:

Note: The parameter was implemented using the `Enabled, Disabled` parameter validate set rather than a switch to mirror the pattern used by the GitHub API.

#### Issues Fixed

- Fixes #390

#### References

- https://docs.github.com/en/rest/repos/repos#update-a-repository

#### Checklist

- [x] You actually ran the code that you just wrote, especially if you did just "one last quick change".
- [x] Comment-based help added/updated, including examples.
- [x] [Static analysis](https://github.com/microsoft/PowerShellForGitHub/blob/master/CONTRIBUTING.md#static-analysis) is reporting back clean.
- [x] New/changed code adheres to our [coding guidelines](https://github.com/microsoft/PowerShellForGitHub/blob/master/CONTRIBUTING.md#coding-guidelines).
- [ ] [Formatters were created](https://github.com/microsoft/PowerShellForGitHub/blob/master/CONTRIBUTING.md#formatters) for any new types being added.
- [x] New/changed code continues to [support the pipeline](https://github.com/microsoft/PowerShellForGitHub/blob/master/CONTRIBUTING.md#pipeline-support).
- [ ] Changes to the manifest file follow the [manifest guidance](https://github.com/microsoft/PowerShellForGitHub/blob/master/CONTRIBUTING.md#module-manifest).
- [x] Unit tests were added/updated and are all passing. See [testing guidelines](https://github.com/microsoft/PowerShellForGitHub/blob/master/CONTRIBUTING.md#testing).  This includes making sure that all pipeline input variations have been covered.
- [ ] Relevant usage examples have been added/updated in [USAGE.md](https://github.com/microsoft/PowerShellForGitHub/blob/master/USAGE.md).
- [ ] If desired, ensure your name is added to our [Contributors list](https://github.com/microsoft/PowerShellForGitHub/blob/master/CONTRIBUTING.md#contributors)

### CODE REVIEW

**Comment by X-Guardian** [comment]
@HowardWolosky, this PR is ready for review.

File: GitHubRepositories.ps1 [hunk 1]
```diff
@@ -1185,6 +1188,9 @@ filter Set-GitHubRepository
 
         [switch] $IsTemplate,
 
+        [ValidateSet('Enabled', 'Disabled')]
+        [string] $SecretScanning,
```
Comment (HowardWolosky) [inline GitHubRepositories.ps1:1192]: My preference is to have all the switch parameters be at the end of the param set (with `$AccessToken` being the final one).  Given that, please move this to be before the first `[switch]` (as well as the `.PARAMETER` documentation as well).

**Review by HowardWolosky — CHANGES_REQUESTED** [review]
Small change requested here, otherwise looks good.  I was a bit hesitant on using `enabled/disabled` here vs a `[switch]`, but after considering it more, it's a bit more clear for this usage.

File: GitHubRepositories.ps1 (hunk 1 above)
Comment (X-Guardian) [reply GitHubRepositories.ps1]: Done

**Comment by X-Guardian** [comment]
@HowardWolosky, any timescale on when you will be able to re-review this PR?

**Review by HowardWolosky — APPROVED** [review]
Thanks for the update (and apologies on the delayed follow-up review).

**Comment by HowardWolosky** [comment]
> /azp run PowerShellForGitHub-CI

It looks like there is a [consistent failure with this change](https://dev.azure.com/ms/PowerShellForGitHub/_build/results?buildId=440824&view=ms.vss-test-web.build-test-results-tab) on two `New-GitHubRepository` tests.  The failures are occurring when validating the `has_wiki` value.  I'm going to kick off a run against `master` to see if the failure is happening without your changes as well.

**Comment by X-Guardian** [comment]
> > /azp run PowerShellForGitHub-CI
> 
> It looks like there is a [consistent failure with this change](https://dev.azure.com/ms/PowerShellForGitHub/_build/results?buildId=440824&view=ms.vss-test-web.build-test-results-tab) on two `New-GitHubRepository` tests. The failures are occurring when validating the `has_wiki` value. I'm going to kick off a run against `master` to see if the failure is happening without your changes as well.

It looks like Wikis aren't supported on private repos on the free tier anymore. I'll raise a PR to update the tests to not check for the `has_wiki` value in these circumstances.

File: GitHubRepositories.ps1 [hunk 2]
```diff
@@ -1251,6 +1257,14 @@ filter Set-GitHubRepository
     if ($PSBoundParameters.ContainsKey('WebCommitSignoffRequired')) { $hashBody['web_commit_signoff_required'] = $WebCommitSignoffRequired.ToBool() }
     if ($PSBoundParameters.ContainsKey('Archived')) { $hashBody['archived'] = $Archived.ToBool() }
 
+    $securityAndAnalysis = @{}
+    if ($PSBoundParameters.ContainsKey('SecretScanning')) {
+        $securityAndAnalysis['secret_scanning'] = @{ 'status' = $SecretScanning.ToLower()}
+    }
+    if ($securityAndAnalysis.count -ne 0) {
```
Comment (HowardWolosky) [inline GitHubRepositories.ps1:1264]:
```suggestion
    if ($securityAndAnalysis.Count -ne 0) {
```

**Review by HowardWolosky — APPROVED** [review]
Ready to go -- thanks for your patience here.   Added a nit as a suggestion (capitalization of the `Count` property) so that I can just apply it and merge it without inconveniencing you any further.

**Pull request merged**